### DISCUSSION TITLE
Improve multiple timing point map's qualify of life

### DISCUSSION THREAD

**Discussion by ServatoryOfficial** [ppy/osu#16831] [Ideas] [answered]
I want to request and discuss some stuff regarding osu! editor mode that's been bugging me while I'm working on map that has a lot of timing points.

1. **Separate the option "Move already placed notes when changing the offset/BPM" to [1st option] "Move already placed notes when the changing the offset" and [2nd option]"Move already placed notes when the changing the BPM". Also turn the [2nd option] (Move already placed notes when the changing the BPM) off when osu! detected the map is using more than 1 timing point by default**

- This is to ensure that maps that have multiple BPM (including the map that has more than 1 timing point despite having the same bpm value) didn't get ruined by the [2nd option]. When the feature is turned on by default, if you're changing the bpm when there are multiple timing points on the map, the rest of the notes on those different timing points will get ruined by this.
Since this feature is "only useable" if the map has a single timing point, there is no need to turn this on by default if the map has multiple timing point. 

2. **Enhance "Resnap all notes" option so it snaps towards the nearest timing point/redline. If there is no timing point near the notes, snap it to the previous snapping that's being used on that certain part**

- [1st problem] Snapping is been something that's been bugging mappers for a while, especially if the map is being mapped while timed at the same time. Resnapping the note is one of the option to quickly move all notes to the correct position. However, this feature performs horribly if the map has multiple timing point that isn't being snapped correctly due to the song provided the mistimed note. 

- [2nd problem] If the map has different snapping (example : 1/8 and 1/12). Mappers have to double check in case he/she accidentally use "Resnap all notes" and currently use 1/8 snapping because the feature only snaps it to 1/2, 1/4 and 1/8. This is really problematic, since the only alternative to counter this problem is to add another timing point that can snap the 1/12 on 1/8 (which I assume no one is gonna do since this will cause the timing to be inconsistent) or do the check manually on that particular area.
The reason of why I didn't ask to **snap it to the nearest snapping available** is due to 1/5, 1/7 and 1/9 snapping (which imo rarely been used) ruining this feature.

3. **Reduce load time when opening the editor if the map has a lot of objects or greenlines/redlines**

- The load time when entering editor mode is usually longer than when the map is loaded to be played. I'm not sure why this happen in the first place, so I cannot provide any suggestion regarding how to fix this issue.

4. **add "do not add cymbal" in the "style" menu (on the Timing Setup Panel) when making a new timing point**

- When mapping a map with multiple timing point, most mapper will try to fulfill the 4/1 Barline (in case the map uses 4/4 time signature) and ignore the Nightcore cymbal issue. This feature might not be noticeable to many mapper, but if implemented correctly, this will probably fix the cymbal position issue if the map has multiple variable timing points. 
In case the timing point is omitted. This feature will automatically checked

**Comment by peppy** [comment, marked as answer]
Most of the points you bring up will not be issues as we don't intend on transferring the features in the way they are implemented in stable to the new editor.

Out of curiosity, have you tried the lazer editor yet? All your issues here are stable issues, not lazer.

**Reply by ServatoryOfficial** [reply]
oh I see, I haven't tried lazer yet tho. I guess I put it on the wrong repo

**Comment by Gyoshi** [comment]
Can someone close this? Since it is not relevant to lazer
